eng/native/configurecompiler.cmake

```diff
@@ -434,6 +434,7 @@ if (CLR_CMAKE_HOST_UNIX)
   elseif(CLR_CMAKE_HOST_NETBSD)
     message("Detected NetBSD amd64")
   elseif(CLR_CMAKE_HOST_SUNOS)
+    add_definitions(-DHOST_SUNOS)
```

Comment:
I think we should just use `TARGET_SUNOS` and not define this. When mapping `host,build,target` unix-y concepts on `host,target`, target is what we actually want 99% of the time, which is why Linux and FreeBSD are still not requiring its definition and those ports are completed since many releases.

Reply:
Yup, HOST_ is mostly used for DAC like stuff or previously in old crossgen1 type of components.
In mono, the logic is more unix-y, host+build is HOST, so it's a fusion between unix-y and windows style, this explains unix-y terminologies: https://stackoverflow.com/a/78396726/863980.